nrfcloud/validatedFetch.spec.ts

```diff
@@ -50,4 +50,41 @@ describe('validatedFetch()', () => {
 			await vf({ resource: 'some-resource' }, mockFetch as any),
 		).toMatchObject({ error: err })
 	})
+
+	it('should override the defaults if init parameter is provided', async () => {
+		const mockFetch = jest.fn(() => ({
+			ok: true,
+			json: async () =>
+				Promise.resolve({
+					foo: 'bar',
+				}),
+		}))
+		const vf = validatedFetch(
+			{
+				endpoint: new URL('https://example.com/'),
+				apiKey: 'some-key',
+			},
+			mockFetch as any,
+		)
+		const schema = Type.Object({ foo: Type.Literal('bar') })
+
+		const res = await vf({ resource: 'foo' }, schema, {
+			method: 'POST',
+			headers: {
+				Authorization: 'Bearer another-key',
+				'Content-Type': 'application/octet-stream',
+			},
+		})
```

Comment:
If refactored this, because this mixes two parameters to configure the fetch behaviour. Those two (the first and the third) are also split around the schema.